ibis/clickhouse/types.py

```diff
@@ -0,0 +1,59 @@
+from toolz import itemmap, merge, pluck
+
+# TODO cleanup, reorganize
+
+_sql_type_names = {
+    'null': 'Null',
+    'int8': 'Int8',
+    'int16': 'Int16',
+    'int32': 'Int32',
+    'int64': 'Int64',
+    'float': 'Float32',
+    'double': 'Float64',
+    'string': 'String',
+    'boolean': 'UInt8',
+    'date': 'Date',
+    'timestamp': 'DateTime',
+    'decimal': 'UInt64',  # see Clickhouse issue #253
+}
+
+
+def _type_to_sql_string(tval):
+    return _sql_type_names[tval.name.lower()]
+
+
+# Nullable(UInt8) etc.
+_dtypes = {'object': 'String',
+           'uint64': 'UInt64',
+           'uint32': 'UInt32',
+           'uint16': 'UInt16',
+           'float64': 'Float64',
+           'float32': 'Float32',
+           'uint8': 'UInt8',
+           'int64': 'Int64',
+           'int32': 'Int32',
+           'int16': 'Int16',
+           'int8': 'Int8',
+           'datetime64[D]': 'Date',
+           'datetime64[ns]': 'DateTime'}
+
+PD2CH = merge(_dtypes, {'bool': 'UInt8'})
+CH2PD = merge(itemmap(reversed, _dtypes), {'Null': 'object'})
```

Comment:
Can you give these longer names?

Reply:
I need a layer between clickhouse-driver and `ibis.clickhouse.api.ClickhouseClient` to read data as a DataFrame, I'm not really satisfied with my current workaround: https://github.com/ibis-project/ibis/pull/1127/files#diff-cb1dd568c3e11460210845acec54bc13R33

Suggestions?